File: contest/gcj-2019-q/pA.py
T = int(input())
for i in range(T):
    n = int(input())
    a = 0
    b = 0
    rate = 1
    idx = 1
    while n > 0:
        ld = n % 10
        n = n // 10
        da = ld // 2
        db = ld - da
        if idx % 2 == 0:
            da,db = db,da

        if da == 4 or db == 4:
            da -= 1 if da < db else -1
            db += 1 if da < db else -1

        a += da * rate
        b += db * rate
        rate *= 10
        idx += 1
    if a == 0:
        a += 1
        b -= 1

    if b == 0:
        a -= 1
        b += 1
    print("Case #{}: {} {}".format(i+1,a,b))

File: contest/CF_1211/pA.kt
private fun readLn() = readLine()!! // string line
private fun readInt() = readLn().toInt() // single int
private fun readStrings() = readLn().split(" ") // list of strings
private fun readInts() = readStrings().map { it.toInt() } // list of ints
 
fun main() {
    var n = readInt()
    var a : IntArray = IntArray(n)

    var mn = 1
    var mx = 1

    var idx = 0;
    for (d in readInts()) {
        a[idx] = d
        idx++;
    }

    for (i in 1..n) {
        if (a[i-1] < a[mn-1]) {
            mn = i;
        }
        if (a[i-1] > a[mx-1]) {
            mx = i;
        }
    }

    var mid = -1
    for (i in 1..n) {
        if (a[i-1] != a[mn-1] && a[i-1] != a[mx-1]) {
            mid = i;
        }
    }

    if (mid == -1) {
        println("-1 -1 -1")
    } else {
        println("%d %d %d".format(mn, mid, mx))
    }
}